### PR TITLE
Add unit tests to noise

### PR DESCRIPTION
**Pull request by schultetwin1**
Introduces some basic encapsulate/decapsulate tests for the noise mod

### CODE REVIEW

**Comment by Noah-Kennedy** [comment]
You have displeased `rustfmt`

**Review by Noah-Kennedy — APPROVED** [review]

**Comment by Noah-Kennedy** [comment]
This is good to merge as soon as you fix `rustfmt`.

File: boringtun/src/noise/timers.rs [hunk 1]
```diff
@@ -6,14 +6,20 @@ use crate::noise::{Tunn, TunnResult};
...
+#[cfg(test)]
+use mock_instant::Instant;
+
+#[cfg(not(test))]
+use std::time::Instant;
 
 // Some constants, represent time in seconds
 // https://www.wireguard.com/papers/wireguard.pdf#page=14
-const REKEY_AFTER_TIME: Duration = Duration::from_secs(120);
+pub const REKEY_AFTER_TIME: Duration = Duration::from_secs(120);
```
Comment (jeff-hiner) [inline boringtun/src/noise/timers.rs:19]: We shouldn't expose any more internals at the lib crate level, especially just for tests. Try `pub(crate)` or `pub(super)`?

File: boringtun/src/noise/timers.rs [hunk 2]
```diff
@@ -6,14 +6,20 @@ use crate::noise::{Tunn, TunnResult};
...
 const REJECT_AFTER_TIME: Duration = Duration::from_secs(180);
 const REKEY_ATTEMPT_TIME: Duration = Duration::from_secs(90);
-const REKEY_TIMEOUT: Duration = Duration::from_secs(5);
+pub const REKEY_TIMEOUT: Duration = Duration::from_secs(5);
```
Comment (jeff-hiner) [inline boringtun/src/noise/timers.rs:22]: Same here.

**Review by jeff-hiner — CHANGES_REQUESTED** [review]

**Comment by schultetwin1** [comment]
> This is good to merge as soon as you fix `rustfmt`.

Fixed!

File: boringtun/src/noise/timers.rs (hunk 1 above)
Comment (schultetwin1) [reply boringtun/src/noise/timers.rs]: Good call, done!

File: boringtun/src/noise/timers.rs (hunk 2 above)
Comment (schultetwin1) [reply boringtun/src/noise/timers.rs]: Done!

**Comment by Noah-Kennedy** [comment]
@schultetwin1 try pushing an empty commit, the integration tests are failing. I'm hoping this is spurious.

**Comment by schultetwin1** [comment]
> @schultetwin1 try pushing an empty commit, the integration tests are failing. I'm hoping this is spurious.

I'm looking into it. I'm wondering if it's not spurious. I did introduce the mocking on `Instant` which could mean time is not increasing.

**Review by jeff-hiner — APPROVED** [review]

**Comment by Noah-Kennedy** [comment]
Yeah that might be doing it.

**Comment by Noah-Kennedy** [comment]
@schultetwin1 you might want to move that out of `#[cfg(test)]` and into some special test CI path, behind a rustc `--cfg` flag. This would allow us to do this without any users being able to accidentally opt into this by enabling features.

**Comment by schultetwin1** [comment]
> @schultetwin1 you might want to move that out of `#[cfg(test)]` and into some special test CI path, behind a rustc `--cfg` flag. This would allow us to do this without any users being able to accidentally opt into this by enabling features.

Good idea, done! I'm pretty sure the tests will now run since we have `cargo hack test --each-feature`

**Pull request merged**